go/vt/wrangler/traffic_switcher.go

```diff
@@ -330,11 +335,13 @@ func (wr *Wrangler) SwitchReads(ctx context.Context, targetKeyspace, workflowNam
 		if servedType != topodatapb.TabletType_REPLICA && servedType != topodatapb.TabletType_RDONLY {
 			return nil, fmt.Errorf("tablet type must be REPLICA or RDONLY: %v", servedType)
 		}
-		if direction == workflow.DirectionBackward && servedType == topodatapb.TabletType_REPLICA && len(ws.ReplicaCellsSwitched) == 0 {
-			return nil, fmt.Errorf("requesting reversal of read traffic for REPLICAs but REPLICA reads have not been switched")
-		}
-		if direction == workflow.DirectionBackward && servedType == topodatapb.TabletType_RDONLY && len(ws.RdonlyCellsSwitched) == 0 {
-			return nil, fmt.Errorf("requesting reversal of SwitchReads for RDONLYs but RDONLY reads have not been switched")
+		if !ts.isPartialMigration { // shard level traffic switching is all or nothing
```

Comment:
This fixes an additional bug:
```
❯ vtctlclient MoveTables -- --source xtra --source_shards='80-' --tables 'customer' ReverseTraffic xtra2.80dash
E0717 10:08:06.507644   59806 main.go:96] E0717 14:08:06.506797 vtctl.go:2188]
requesting reversal of SwitchReads for RDONLYs but RDONLY reads have not been switched
I0717 10:08:06.514211   59806 main.go:96] I0717 14:08:06.514090 vtctl.go:2190] Workflow Status: Reads partially switched, for shards: 80-. Writes partially switched, for shards: 80-

The following vreplication streams exist for workflow xtra2.80dash:

id=1 on 80-/zone1-0000000330: Status: Stopped. VStream Lag: 0s.

MoveTables Error: rpc error: code = Unknown desc = requesting reversal of SwitchReads for RDONLYs but RDONLY reads have not been switched'
```

I had seen that in the unit test. Confirmed with a local test. The problem is that we were processing those values even for partial workflows, when they are not ever set for partial workflows: https://github.com/vitessio/vitess/blob/d7704d4a579adb71d15349c7aa380cf907cbf2c8/go/vt/wrangler/traffic_switcher.go#L247-L267